test/hotspot/jtreg/runtime/ErrorHandling/TestHeapDumpOnOutOfMemoryAndCrashOnOutOfMemory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test TestHeapDumpOnOutOfMemoryAndCrashOnOutOfMemory
+ * @summary Test verifies call to -XX:HeapDumpOnOutOfMemoryError and
+ *          CrashOnOutOfMemoryError handled in a single safepoint operation
+ * @library /test/lib
+ * @requires vm.flagless
+ * @run driver TestHeapDumpOnOutOfMemoryAndCrashOnOutOfMemory
+ */
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class TestHeapDumpOnOutOfMemoryAndCrashOnOutOfMemory {
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 1) {
+            try {
+                Object[] oa = new Object[Integer.MAX_VALUE];
+                for(int i = 0; i < oa.length; i++) {
+                    oa[i] = new Object[Integer.MAX_VALUE];
```

Comment:
This will throw the "VM limit reached" OOME - does that trigger the heapdump etc processing?

Reply:
I have copied this code from TestHeapDumpOnOutOfMemoryError.java, and I have restricted heap to -Xmx128M to force OOM in heap.

Reply:
The heap size will make no difference as you are trying to create an array that is larger than the VM allows. Line 42 will throw OOME and the for loop is never reached.
```
|  Welcome to JShell -- Version 23
|  For an introduction type: /help intro

jshell> Object[] oa = new Object[Integer.MAX_VALUE];
|  Exception java.lang.OutOfMemoryError: Requested array size exceeds VM limit
|        at (#1:1)
```

Reply:
I also observed that in Jshell gives `Requested array size exceeds VM limit` , but during program execution, it gets into this 

`STDERR:
 stdout: [java.lang.OutOfMemoryError: Java heap space

Dumping heap to java_pid238354.hprof ...

Heap dump file created [187789641 bytes in 0.285 secs]

Aborting due to java.lang.OutOfMemoryError: Java heap space
`

Based on this assumption only below test are defined.
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/runtime/ErrorHandling/TestHeapDumpOnOutOfMemoryError.java
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/runtime/ErrorHandling/TestHeapDumpPath.java
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/runtime/ErrorHandling/TestGZippedHeapDumpOnOutOfMemoryError.java